apps/prairielearn/src/components/Navbar.html.ts

```diff
@@ -0,0 +1,853 @@
+                `,
+              )}
+            </div>
+          </div>
+        `
+      : ''}
+
+    <a class="dropdown-item" href="${effectiveUserUrl}">Customize&hellip;</a>
+
+    <div class="dropdown-divider"></div>
+  `;
+}
+
+function NavbarPlain({ resLocals, navPage }: { resLocals: Record<string, any>; navPage: NavPage }) {
+  const { is_administrator } = resLocals;
+  if (!is_administrator) return '';
+  return html`
+    <li class="nav-item ${navPage === 'admin' ? 'active' : ''}">
+      <a class="nav-link" href="${config.urlPrefix}/administrator/admins">Admin</a>
+    </li>
+  `;
+}
+
+function NavbarStudent({
+  resLocals,
+  navPage,
+}: {
+  resLocals: Record<string, any>;
+  navPage: NavPage;
+}) {
+  const { course, course_instance, assessment_instance, assessment_instance_label, urlPrefix } =
+    resLocals;
+
+  return html`
+    <li class="nav-item navbar-text mr-4">
+      ${course?.short_name ?? ''}, ${course_instance?.short_name ?? ''}
+    </li>
+
+    <li class="nav-item ${navPage === 'assessments' ? 'active' : ''}">
+      <a class="nav-link" href="${urlPrefix}/assessments">Assessments</a>
+    </li>
+    <li class="nav-item ${navPage === 'gradebook' ? 'active' : ''}">
+      <a class="nav-link" href="${urlPrefix}/gradebook">Gradebook</a>
+    </li>
+
+    ${assessment_instance_label != null && assessment_instance != null
+      ? html`
+          <li class="nav-item ${navPage === 'assessment_instance' ? 'active' : ''}">
+            <a class="nav-link" href="${urlPrefix}/assessment_instance/${assessment_instance.id}">
+              ${assessment_instance_label}
+            </a>
+          </li>
+        `
+      : ''}
+  `;
+}
+
+function NavbarInstructor({
+  resLocals,
+  navPage,
+  navSubPage,
+}: {
+  resLocals: Record<string, any>;
+  navPage: NavPage;
+  navSubPage?: NavSubPage;
+}) {
+  const {
+    course,
+    course_instance,
+    assessment,
+    assessment_label,
+    assessments,
+    navbarOpenIssueCount,
+    authz_data,
+    urlPrefix,
+  } = resLocals;
+
+  return html`
+    <li class="nav-item btn-group" id="navbar-course-switcher">
+      <a
+        class="nav-link ${navPage === 'course_admin' &&
+        !(navSubPage === 'issues' || navSubPage === 'questions' || navSubPage === 'syncs')
+          ? 'active'
+          : ''} ${!authz_data.has_course_permission_view ? 'disabled' : ''}"
+        href="${urlPrefix}/course_admin"
+      >
+        ${course.short_name}
+      </a>
+      <a
+        class="nav-link dropdown-toggle dropdown-toggle-split"
+        href="#"
+        id="navbarDropdownMenuCourseAdminLink"
+        role="button"
+        data-toggle="dropdown"
+        aria-label="Change course"
+        aria-haspopup="true"
+        aria-expanded="false"
+        ${!authz_data.overrides
+          ? html`
+              hx-get="/pl/navbar/course/${course.id}/switcher" hx-trigger="show-course-switcher once
+              delay:200ms" hx-target="#navbarDropdownMenuCourseAdmin"
```

Comment:
Another case where the aforementioned `htmlAttributes` utility could be useful 👀